packages/nitro-protocol/contracts/examples/RockPaperScissorsGame.sol

```diff
@@ -0,0 +1,295 @@
+                validateRoundProposedToRoundAccepted(a, b);
+
+                return true;
+            }
+        } else if (fromGameData.positionType == PositionType.RoundAccepted) {
+            if (toGameData.positionType == PositionType.Reveal) {
+                validateRoundAcceptedToReveal(a, b);
+
+                return true;
+            }
+        } else if (fromGameData.positionType == PositionType.Reveal) {
+            if (toGameData.positionType == PositionType.Start) {
+                validateRevealToStart(a, b);
+
+                return true;
+            }
+        }
+
+        revert('No valid transition found for commitments');
+    }
+
+    function winnings(
+        Weapon playerFirstWeapon,
+        Weapon playerSecondWeapon,
+        uint256 stake
+    ) private pure returns (uint256, uint256) {
+
+        if (playerFirstWeapon == playerSecondWeapon) {
+            return (stake, stake);
+        } else if (
+            (playerFirstWeapon == Weapon.Rock && playerSecondWeapon == Weapon.Scissors) ||
+            (playerFirstWeapon > playerSecondWeapon)
+        ) {
+            // first player won
+            return (2 * stake, 0);
+        } else {
+            // second player won
+            return (0, 2 * stake);
+        }
+    }
+
+    modifier outcomeUnchanged(VariablePart memory a, VariablePart memory b) {
+        require(keccak256(b.outcome) == keccak256(a.outcome), 'RockPaperScissorsGame: Outcome must not change');
+        _;
+    }
+
+    modifier allocationUnchanged(
+        VariablePart memory a,
+        VariablePart memory b
+    ) {
+        Outcome.OutcomeItem[] memory outcomeA = abi.decode(a.outcome, (Outcome.OutcomeItem[]));
+        Outcome.OutcomeItem[] memory outcomeB = abi.decode(b.outcome, (Outcome.OutcomeItem[]));
+        Outcome.AssetOutcome memory assetOutcomeA = abi.decode(outcomeA[0].assetOutcomeBytes, (Outcome.AssetOutcome));
+        Outcome.AssetOutcome memory assetOutcomeB = abi.decode(outcomeB[0].assetOutcomeBytes, (Outcome.AssetOutcome));
+        Outcome.AllocationItem[] memory allocationA = abi.decode(
+            assetOutcomeA.allocationOrGuaranteeBytes,
+            (Outcome.AllocationItem[])
+        );
+        Outcome.AllocationItem[] memory allocationB = abi.decode(
+            assetOutcomeB.allocationOrGuaranteeBytes,
+            (Outcome.AllocationItem[])
+        );
+        require(allocationB[0].destination == allocationA[0].destination,'RockPaperScissorsGame: Destimation playerFirst may not change');
+        require(allocationB[1].destination == allocationA[1].destination,'RockPaperScissorsGame: Destimation playerSecond may not change');
+        require(allocationB[0].amount == allocationA[0].amount,'RockPaperScissorsGame: Amount playerFirst may not change');
+        require(allocationB[1].amount == allocationA[1].amount,'RockPaperScissorsGame: Amount playerSecond may not change');
+        _;
+    }
+
+    modifier oneAssetType(
```

Comment:
We should consider putting these in a library for app developers to reuse (they might have to be functions rather than modifiers, then).